src/components/Tool/FixCommitmentInfo/FixCommitmentInfo.tsx

```diff
@@ -65,122 +56,185 @@ const useStyles = makeStyles()((theme: Theme) => ({
+      return {};
+    }
+
+    try {
+      return JSON.parse(suggestedChanges);
+    } catch (error) {
+      return {};
+    }
+  };
+
+  const updateContact = async (): Promise<void> => {
+    let attributes;
+
+    switch (modalState.updateType) {
+      case 'CHANGE':
+        attributes = {
+          id: modalState.contact.id,
+          status: modalState.contact.status,
+          pledgeAmount: modalState.contact.pledgeAmount,
+          pledgeCurrency: modalState.contact.pledgeCurrency,
+          pledgeFrequency: modalState.contact.pledgeFrequency,
+          statusValid: true,
+        };
+        break;
+      case 'DONT_CHANGE':
+        attributes = {
+          id: modalState.contact.id,
           statusValid: true,
-        }
-      : { id, statusValid: true };
+        };
+        break;
+      case 'HIDE':
+        attributes = {
+          id: modalState.contact.id,
+          status: 'NEVER_ASK' as StatusEnum,
+        };
+        break;
+    }
+
     await updateInvalidStatus({
       variables: {
         accountListId,
         attributes,
       },
+      update: (cache) => {
+        cache.evict({ id: `Contact:${modalState.contact.id}` });
+      },
+      onError() {
+        enqueueSnackbar(
+          t(`Error updating ${modalState.contact.name}'s commitment info`),
+          {
+            variant: 'error',
+            autoHideDuration: 7000,
+          },
+        );
+      },
+      onCompleted() {
+        enqueueSnackbar(
+          t(`${modalState.contact.name}'s commitment info updated!`),
+          {
+            variant: 'success',
+            autoHideDuration: 7000,
+          },
+        );
+      },
     });
-    enqueueSnackbar(t('Contact commitment info updated!'), {
-      variant: 'success',
-    });
-    hideContact(id);
   };
 
-  const hideContact = (hideId: string): void => {
-    const query = {
-      query: GetInvalidStatusesDocument,
-      variables: {
-        accountListId,
-      },
-    };
-
-    const dataFromCache = client.readQuery<GetInvalidStatusesQuery>(query);
-
-    if (dataFromCache) {
-      const data = {
-        ...dataFromCache,
-        contacts: {
-          ...dataFromCache.contacts,
-          nodes: dataFromCache.contacts.nodes.filter(
-            (contact) => contact.id !== hideId,
-          ),
-        },
-      };
-
-      client.writeQuery({ ...query, data });
-    }
+  const handleShowModal = (
+    contact: ContactType,
+    message: string,
+    title: string,
+    updateType: UpdateTypeEnum,
+  ) => {
+    setModalState({
+      open: true,
+      contact,
+      message,
+      title,
+      updateType,
+    });
   };
 
   return (
     <Box className={classes.outer} data-testid="Home">
-      {!loading && !updating && !loadingStatuses && data ? (
-        <Grid container className={classes.container}>
+      {!updating && data ? (
+        <Grid container className={classes.container} data-testid="Container">
```

Comment:
Can we remove this `!updating` check so that the list doesn't disappear while you're updating a contact?